include/nbl/builtin/hlsl/scene/animtation.hlsl

```diff
@@ -0,0 +1,42 @@
+
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_HLSL_SCENE_ANIMATION_INCLUDED_
+#define _NBL_HLSL_SCENE_ANIMATION_INCLUDED_
+
+
+#include <nbl/builtin/hlsl/math/keyframe.hlsl>
+
+namespace nbl
+{
+	namespace hlsl
+	{
+		namespace scene
+		{
+			struct Animation_t
+			{
+				uint keyframeOffset; // same offset for timestamps and keyframes
+				uint keyframesCount_interpolationMode; // 2 bits for interpolation mode

```

Comment:
when you see something like this its supposed to be
```cpp
				uint keyframesCount : 30;
				uint interpolationMode : 2;
```

however don't do it yet cause you need a DXC.exe that does not complain about `-HV 2021`

Reply:
or you actually could do that but all the testing you'd need to do with [Compiler Explorer for now
](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:hlsl,selection:(endColumn:2,endLineNumber:9,positionColumn:2,positionLineNumber:9,selectionStartColumn:2,selectionStartLineNumber:9,startColumn:2,startLineNumber:9),source:'namespace+nbl%0A%7B%0Anamespace+hlsl%0A%7B%0A%0Ausing+uint32_t+%3D+uint%3B%0A%0Anamespace+math%0A%7B%0A%0Astruct+quaternion_t%0A%7B%0A%7D%3B%0A%0A%7D%0A%7D%0A%7D%0A%0Anamespace+nbl%0A%7B%0Anamespace+hlsl%0A%7B%0Anamespace+scene%0A%7B%0A%0Ausing+namespace+math%3B%0A++++%0Astruct+Animation_t%0A%7B%0A++++//+same+offset+for+timestamps+and+keyframes%0A++++uint32_t+keyframeOffset%3B%0A++++uint32_t+keyframesCount+:+30%3B%0A++++uint32_t+interpolationMode+:+2%3B%0A%0A++++quaternion_t+stuff()%0A++++%7B%0A++++++++quaternion_t+retval%3B%0A++++++++return+retval%3B%0A++++%7D+%0A%7D%3B%0A%0A%7D%0A%7D%0A%7D'),l:'5',n:'0',o:'HLSL+source+%231',t:'0')),k:51.88383045525903,l:'4',m:100,n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:dxc_trunk,deviceViewOpen:'1',filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:12,fontUsePx:'0',j:1,lang:hlsl,libs:!(),options:'-HV+2021+-T+lib_6_7',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+DXC+(trunk)+(Editor+%231)',t:'0')),k:48.116169544740984,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'DXC+(trunk)',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+DXC+(trunk)+(Compiler+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:48.116169544740984,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)